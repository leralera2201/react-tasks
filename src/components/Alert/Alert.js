import React from "react";
import { Alert } from "reactstrap";

import "./Alert.scss";

const AlertItem = ({ color, message }) => {
  return <Alert color={color}>{message}</Alert>;
};

export default AlertItem;
