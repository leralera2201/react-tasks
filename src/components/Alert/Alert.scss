.alert {
  margin: 0 !important;
}

.Toastify {
  &__close-button {
    position: absolute;
    right: 5px;
  }
  &__toast {
    padding: 0;
    min-height: 0;
    &-body {
      margin: 0;
    }
  }
}
