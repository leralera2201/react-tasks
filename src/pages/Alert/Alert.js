import Alert from "@components/Alert";

const AlertPage = () => {
  return (
    <>
      <Alert type="success" message="Success Alert" showIcon />
      <Alert type="error" message="Error Alert" showIcon />
      <Alert type="info" message="Info Alert" />
      <Alert type="warning" message="Warning Alert" />
    </>
  );
};

export default AlertPage;
