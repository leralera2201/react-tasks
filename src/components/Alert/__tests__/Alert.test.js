import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Alert from "../Alert";

describe("Testing Alert component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Alert message="Alert" color="danger" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
