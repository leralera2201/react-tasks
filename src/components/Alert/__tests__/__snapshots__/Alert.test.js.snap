// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Alert component renders correctly 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  Alert
</Alert>
`;
